metadata/metadata_test.go

```diff
@@ -58,10 +69,13 @@ func TestFilterURL(t *testing.T) {
 		{"/0.1/meta-data/service-accounts/default/identity", concealedErr, ""},
 		{"/computeMetadata/v1beta1/instance/service-accounts/default/identity", concealedErr, ""},
 		{"/computeMetadata/v1/instance/service-accounts/default/identity", concealedErr, ""},
+		{"/computeMetadata/v1/instance/service-accounts/default/identity?audience=www.example.com&format=full", concealedErr, ""},
 		// Recursive (non-whitelisted).
 		{"/computeMetadata/v1/instance/?recursive=true", recursiveErr, ""},
-		{"/computeMetadata/v1/instance/?something=other&recursive=true", recursiveErr, ""},
-		{"/computeMetadata/v1/instance/?recursive=true&something=other", recursiveErr, ""},
+		{"/computeMetadata/v1/instance/?%72%65%63%75%72%73%69%76%65=true", recursiveErr, ""}, // url-hex-encoded
+		{"/computeMetadata/v1/instance/?alt=text&recursive=true", recursiveErr, ""},
+		{"/computeMetadata/v1/instance/?recursive=true&alt=text", recursiveErr, ""},
+		{"/computeMetadata/v1/instance/?alt=text;recursive=true", recursiveErr, ""},
```

Comment:
Can we add a test for `/computeMetadata/v1/instance/?recursive`

Reply:
done